src/coreclr/src/vm/callingconvention.h

```diff
@@ -1550,6 +1550,14 @@ void ArgIteratorTemplate<ARGITERATOR_BASE>::ComputeReturnFlags()
             }
 #endif
 
+#if defined(_TARGET_AMD64_) && !defined(UNIX_AMD64_ABI)
+            if (thValueType.AsMethodTable()->GetVectorSize() == 16) /* (Andrew Au) Check if it is a Vector128<T> */
```

Comment:
This will catch `Vector<T>` as well on non-AVX SSE supporting hardware, , which is awkward as I don't think that's plumbed through everywhere in this change. @CarolEidt, I'm assuming that we treat a `Vector<T>` as a __m128 when producing code which uses it as a 128 bit vector.

Reply:
Yes, `Vector<T>` should be treated as a `__m128` on non-AVX2 hardware. This needs to distinguish that case from `Vector4`.